yolo/utils/metrics.py

```diff
@@ -91,6 +92,10 @@ def ap_per_class(tp, conf, pred_cls, target_cls, plot=False, fname='precision-re
                 ap[ci, j], mpre, mrec = compute_ap(recall[:, j], precision[:, j])
                 if j == 0:
                     py.append(np.interp(px, mrec, mpre))  # precision at mAP@0.5
+            
+            # remap precision and recall curves at iou .5 from objectness to confidence thresholds
+            remapped_precision, remapped_recall = np.interp(np.flip(-px), -conf[i], precision[:, 0]),np.interp(np.flip(-px), -conf[i], recall[:, 0])
```

Comment:
Not sure why you are interpolating here, can't you just read in the curves corresponding to the 0.5 IoU threshold directly?